client/src/test/java/org/eclipse/hono/client/impl/EventConsumerImplTest.java

```diff
@@ -107,4 +107,42 @@ public void testCreateRegistersBiConsumerAsMessageHandler(final TestContext ctx)
         verify(delivery).disposition(any(Released.class), eq(Boolean.TRUE));
     }
 
+    /**
+     * Verifies that the close on receiver calls the closehook.
+     *
+     * @param ctx The test context.
+     */
+    @SuppressWarnings("unchecked")
+    @Test
+    public void testCloseHookOnDetach(final TestContext ctx) {
+
+        // GIVEN an open event consumer
+        BiConsumer<ProtonDelivery, Message> eventConsumer = mock(BiConsumer.class);
+        ProtonReceiver receiver = mock(ProtonReceiver.class);
+        ProtonConnection con = mock(ProtonConnection.class);
+        when(con.createReceiver(anyString())).thenReturn(receiver);
+        when(receiver.isOpen()).thenReturn(Boolean.TRUE);
+        when(receiver.openHandler(any(Handler.class))).thenAnswer(invocation -> {
+            invocation.getArgumentAt(0, Handler.class).handle(Future.succeededFuture(receiver));
+            return receiver;
+        });
+        when(receiver.closeHandler(any(Handler.class))).thenAnswer(invocation -> {
+            invocation.getArgumentAt(0, Handler.class).handle(Future.succeededFuture(receiver));
+            return receiver;
+        });
+        Handler<String> closeHook = mock(Handler.class);
+        Async consumerCreation = ctx.async();
+
+        // WHEN the receiver link is closed
+        EventConsumerImpl.create(vertx.getOrCreateContext(), new ClientConfigProperties(), con, "tenant", eventConsumer, ctx.asyncAssertSuccess(s -> {
+            s.close(closeHandler -> {
+                consumerCreation.complete();
+            });
+        }), closeHook );
+        consumerCreation.await(500);
+
+        // THEN the close hook is called
+        verify(closeHook).handle(any());
```

Comment:
analogously to the test case for `ForwardingDownstreamAdapter` I would propose something like this:

```
...
when(receiver.isOpen()).thenReturn(Boolean.TRUE);
Handler<String> closeHook = mock(Handler.class);
ArgumentCaptor<Handler> detachCaptor = ArgumentCaptor.forClass(Handler.class);
Async consumerCreation = ctx.async();
EventConsumerImpl.create(vertx.getOrCreateContext(), new ClientConfigProperties(), con, "tenant", eventConsumer, 
       ctx.asyncAssertSuccess(s -> consumerCreation.complete()), closeHook );
consumerCreation.await(500);
verify(receiver).detachHandler(detachCaptor.getCapture());

// WHEN the receiver link is closed
detachCaptor.getValue().handle(..);

// THEN the close hook is called
verify(closeHook).handle(any());
// and the receiver link is closed
verify(receiver).close();
```